### PR TITLE
Feature color legend component

### PR DESCRIPTION
**Pull request by mitchthorson**
### What's in this pull request

- [x] New component/feature

### Description

A companion ColorLegend component to pair with the SVGMap. Takes a D3 color scale and does its best to render a legend with sensible defaults and options for customization.

### Before submitting, please check that you've

- [x] Formatted your code correctly (i.e., prettier cleaned it up)
- [x] Documented any new components or features
- [x] Added any changes in this PR to the `CHANGELOG.md` `Next` section
- [x] If this pull request includes a new component or feature, has it been exported from one of the library's entry points?

### CODE REVIEW

File: src/lib/maps/ColorLegend/ColorLegend.svelte [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+                    stroke-width={tickLineWidth}
+                  ></line>
+                {/if}
+                <text
+                  fill="black"
+                  alignment-baseline={tickPosition === "top" ? "hanging" : "baseline"}
+                  y={yPosition}
+                  x={xPosition}
+                  font-size="{tickSize}px"
+                  text-anchor="middle">{tickFormatFn(tick)}</text
```
Comment (benkates) [inline src/lib/maps/ColorLegend/ColorLegend.svelte:319]: consider adding a prop/some sort of config for modifying `text-anchor`'s value. "end" for last value and "start" for first, suggestion below would require adding an index variable `i` in `{#each}`
```suggestion
text-anchor={i === legendTicks.length ? "end" : i === 0 ? "start" : "middle"}>{tickFormatFn(tick)}</text
```

**Comment by benkates** [comment]
Some non-code specific comments here, feature considerations, etc.:
- In non-diverging scales, I'm unable to have 0 appear at the beginning of the scale. What are your thoughts here?
- In Datawrapper, sometimes I have cases where it's easier for the user to read a custom swatch-like scale based on linear data like below. Right now when I try to set `swatch` to true with a threshold/linear scale it still appears as a bar. This can certainly be a future state thing, but figured I'd bring it up!
image
- Related/similar to the above, in threshold bars it'd be nice to show the range as the label in the middle of a bar section. That has always been my preferred way of reading legends! (I suppose this is just the swatch layout)
- Feature consideration: vertical layout for bars (non-swatch)

File: src/lib/maps/ColorLegend/ColorLegend.svelte (hunk 1 above)
Comment (mitchthorson) [reply src/lib/maps/ColorLegend/ColorLegend.svelte:319]: This is a good call. I might need to try something a bit more robust than just the first/last tick defaulting to "end" and "start". There are some cases where that wouldn't be desirable (cases where the first tick doesn't align with the min of the scale and where the last tick doesn't align with the max of the scale) but we should definitely expose a bit more control as well as some better defaults.

**Comment by mitchthorson** [comment]
@benkates thanks for these. A few things here that can be addressed with some better documentation (how to manually specify tick marks if the D3 algorithm doesn't give you what you want), a few things that could be good feature additions (vertical layout for non-swatch scales would be some low-hanging fruit for sure). In terms of the legend design, I have some thoughts on the range labeling and best practices there that may beyond the scope of this component, let's talk more about those BUT i think there are ways to achieve the effect you described by using the ordinal scale/swatch option by pre-assigning buckets to your data.

**Comment by mitchthorson** [comment]
@benkates added some better handling for `text-anchor` including a property to override. Also added some examples for custom `tickValues` as a way to ensure you see the ticks you need and a way of using the ordinal scale to show bucket labels.

**Comment by mitchthorson** [comment]
Moving these changes to a unified PR: #89

**Pull request closed without merging**